addons/mrp/models/mrp_workcenter.py

```diff
@@ -154,9 +155,26 @@ def _compute_productive_time(self):
 
     @api.depends('blocked_time', 'productive_time')
     def _compute_oee(self):
+        time_data = self.env['mrp.workcenter.productivity'].read_group(
+            domain=[
+                ('date_start', '>=', fields.Datetime.to_string(datetime.now() - relativedelta.relativedelta(months=1))),
+                ('workcenter_id', 'in', self.ids),
+                ('date_end', '!=', False),
+            ],
+            fields=['workcenter_id', 'loss_type', 'duration'],
+            groupby=['workcenter_id', 'loss_type'],
+            lazy=False,
+        )
```

Comment:
```suggestion
        time_data = self.env['mrp.workcenter.productivity']._read_group(
            domain=[
                ('date_start', '>=', fields.Datetime.to_string(datetime.now() - relativedelta.relativedelta(months=1))),
                ('workcenter_id', 'in', self.ids),
                ('date_end', '!=', False),
            ],
            groupby=['workcenter_id', 'loss_type'],
            aggregates=['duration:sum'],
        )
```
As a rule of thumb, always prefer `_read_group`. More details on the *why* [here](https://github.com/odoo/odoo/pull/110737).
And this way you don't need to sum the `duration` afterwards either.